bgpd/bgp_nht.c

```diff
@@ -117,24 +117,36 @@ static int bgp_isvalid_nexthop_for_mplsovergre(struct bgp_nexthop_cache *bnc,
 static int bgp_isvalid_nexthop_for_mpls(struct bgp_nexthop_cache *bnc,
 					struct bgp_path_info *path)
 {
+	return (bnc && (bnc->nexthop_num > 0 &&
+			(CHECK_FLAG(path->flags, BGP_PATH_ACCEPT_OWN) ||
+			 CHECK_FLAG(bnc->flags, BGP_NEXTHOP_LABELED_VALID) ||
+			 bgp_isvalid_nexthop_for_ebgp(bnc, path) ||
+			 bgp_isvalid_nexthop_for_mplsovergre(bnc, path))));
+}
+
+static int bgp_isvalid_nexthop_for_l3vpn(struct bgp_nexthop_cache *bnc,
```

Comment:
nit: bool (I know you want to keep the format as previous ones, but they should be changed too - not with this PR).

Reply:
Fixed, Thanks